packages/phone-number-privacy/signer/src/pnp/endpoints/sign/action.ts

```diff
@@ -89,24 +89,33 @@ export function pnpSign(
     }
 
     let signature: string
-    try {
-      signature = await sign(
-        request.body.blindedQueryPhoneNumber,
-        key,
-        keyProvider,
-        response.locals.logger
-      )
-    } catch (err) {
-      response.locals.logger.error({ err }, 'catch error on signing')
-
-      return errorResult(500, ErrorMessage.SIGNATURE_COMPUTATION_FAILURE, {
-        performedQueryCount: usedQuota,
-        totalQuota: account.pnpTotalQuota,
-      })
+    if (duplicateRequest && duplicateRequest.signature.length) {
+      signature = duplicateRequest.signature
+    } else {
+      try {
+        signature = await sign(
```

Comment:
Would it make sense to `recordRequest()` here, to update the duplicated request with it's signature?

Reply:
We recordRequest below, so I assume you're talking about the edge case where there is a duplicate request but it doesn't have a signature yet. I think because we're going to be pruning the database of old requests, this edge case will never happen in practice after the first time we prune old requests. I'd be in favor of not complicating the code for that reaseon